### PR TITLE
Descriptions: Update content:string to value: { text }

### PR DESCRIPTION
**Pull request by sorja**
part of #1544 

Done:

- [x] update Descriptions (CommentableDescription) type and db table: rename content property to value and db type jsonb and ts type is {text:string,dataSources?:Array<DataSource>} where DataSource = {reference:string,type:string,fraVariable:string,year:string,comments:string}
- [x] update data migration to migrate current content into value ({text})
- [x] FE/BE update how data source is saved and fetched with new format 

image

### CODE REVIEW

File: src/client/pages/AssessmentSection/Descriptions/CommentableDescription/CommentableDescription.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react'
 
 import { Objects } from '@utils/objects'
 
+import { CommentableDescriptionValue } from '@meta/assessment/commentableDescription'
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/Descriptions/CommentableDescription/CommentableDescription.tsx]: Export and import from meta/assessment

File: src/client/pages/AssessmentSection/Descriptions/Description/Description.tsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import './Description.scss'
 import React, { useCallback, useEffect, useState } from 'react'
 
+import { CommentableDescriptionValue } from '@meta/assessment/commentableDescription'
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/Descriptions/Description/Description.tsx]: As above

File: src/client/pages/AssessmentSection/Descriptions/Description/Description.tsx [hunk 3]
```diff
@@ -97,7 +99,7 @@ const Description: React.FC<Props> = (props) => {
 
 Description.defaultProps = {
   disabled: false,
-  template: null,
+  template: { text: null },
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/Descriptions/Description/Description.tsx]: Why sometimes empty string and here null ?

File: src/client/store/pages/assessmentSection/actions/getDescription.ts [hunk 4]
```diff
@@ -1,22 +1,23 @@
-import { ApiEndPoint } from '@meta/api/endpoint'
 import { createAsyncThunk } from '@reduxjs/toolkit'
 import axios from 'axios'
 
+import { ApiEndPoint } from '@meta/api/endpoint'
 import { CycleDataParams } from '@meta/api/request'
+import { CommentableDescriptionValue } from '@meta/assessment/commentableDescription'
```
Comment (minotogna) [inline src/client/store/pages/assessmentSection/actions/getDescription.ts]: Same here , fix it everywhere and I'll keep reviewing

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/meta/assessment/index.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ export { AssessmentNames } from './assessmentName'
 export type { Col, ColProps, ColSelectOption, ColSelectProps, ColStyle } from './col'
 export { ColType } from './col'
 export { Cols } from './cols'
+export type { CommentableDescriptionValue } from './commentableDescription'
```
Comment (minotogna) [inline src/meta/assessment/index.ts]: Export also DataSource and CommentableDescription

File: src/server/api/cycleData/descriptions/upsertDescription.ts [hunk 6]
```diff
@@ -1,20 +1,20 @@
 import { Response } from 'express'
 
 import { CycleDataRequest } from '@meta/api/request'
-import { CommentableDescriptionName } from '@meta/assessment/commentableDescription'
+import { CommentableDescriptionName, CommentableDescriptionValue } from '@meta/assessment/commentableDescription'
```
Comment (minotogna) [inline src/server/api/cycleData/descriptions/upsertDescription.ts]: Import from meta assessment pleaseeeee

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/meta/assessment/index.ts [hunk 7]
```diff
@@ -9,6 +9,12 @@ export { AssessmentNames } from './assessmentName'
 export type { Col, ColProps, ColSelectOption, ColSelectProps, ColStyle } from './col'
 export { ColType } from './col'
 export { Cols } from './cols'
+export type {
+  CommentableDescription,
+  CommentableDescriptionName,
```
Comment (minotogna) [inline src/meta/assessment/index.ts]: shouldn't enum be exported without type ?

File: src/server/repository/assessment/assessment/getCreateSchemaDDL.ts [hunk 8]
```diff
@@ -217,18 +217,15 @@ export const getCreateSchemaCycleDDL = (assessmentSchemaName: string, assessment
...
           country_iso  varchar(3)      not null
               constraint table_name_country_country_iso_fk
                   references country
                   on update cascade on delete cascade,
           section_name varchar(50)     not null,
           name         varchar(50)     not null,
-          content      text default '' not null,
-          constraint table_name_pk_2
-              unique (country_iso, section_name, name)
+          value      jsonb default '{}'::jsonb not null,
```
Comment (minotogna) [inline src/server/repository/assessment/assessment/getCreateSchemaDDL.ts:227]: better to have default '{"text":""}' ?

**Review by minotogna — APPROVED** [review]

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**